rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -167,7 +167,20 @@ impl ArrowBlockfileWriter {
         } else {
             let (split_key, new_delta) = delta.split::<K, V>();
             self.sparse_index.add_block(split_key, new_delta.id);
-            new_delta.add(prefix, key, value);
+            match new_delta.get_min_key() {
+                Some(min_key) => {
+                    if search_key >= min_key {
+                        new_delta.add(prefix, key, value);
+                    } else {
+                        delta.add(prefix, key, value);
+                    }
+                }
+                None => {
+                    // Shouldn't happen but if in any case our MAX_BLOCK_SIZE
```

Comment:
```suggestion
                    // Shouldn't happen but if in any case our MAX_BLOCK_SIZE
```

hmm i would prefer if we didn't handle invariant violating logic like this. We should instead panic imo